symja_android_library/matheclipse-core/src/main/java/org/matheclipse/core/reflection/system/ListLinePlot3D.java

```diff
@@ -29,140 +29,154 @@ public IExpr evaluate(final IAST ast, EvalEngine engine) {
+      // case 4: multiple line coordinates
+      // e.g.: ListLinePlot3D[{{coord1, coord2}, {coord3, coord4}}]
+      if (dimension != null && dimension.length == 2 && dimension[1] == 3) {
+        IASTAppendable result = F.Graphics3D(coordinateLinePlot((IAST) ast.arg1(), plotStyle, engine));
         if (ast.argSize() > 1) {
           // add same options to Graphics3D
           result.appendAll(ast, 2, ast.size());
         }
         return result;
       }
-    }
 
+      // the code reach here, NIL is returned
+    }
     return F.NIL;
-
   }
 
-  private IExpr matrixLinePlot(IExpr heightValueMatrix, int[] dimension, IAST plotStyle,
-      int[] lineColorNumber, double deltaHeight, EvalEngine engine) {
-    // convert possible sparse array expression:
-    IAST values = (IAST) heightValueMatrix.normal(false);
-    if (dimension[0] > 3) {
-      if (dimension[1] == 3) {
-        double minX = (((IAST) values.arg1()).arg1()).evalDouble();
-        double maxX = minX;
-
-        double minY = (((IAST) values.arg1()).arg2()).evalDouble();
-        double maxY = minY;
-
-        double minZ = (((IAST) values.arg1()).arg3()).evalDouble();
-        double maxZ = minZ;
-
-        final IAST color = GraphicsFunctions.plotStyleColorExpr(lineColorNumber[0]++, plotStyle);
-        for (int i = 1; i <= values.argSize(); i++) {
-          IAST row = ((IAST) values.get(i));
-
-          try {
-            // evalDouble may throw ArgumentTypeException
-            double x = row.arg1().evalDouble();
-            if (x < minX)
-              minX = x;
-            if (x > maxX)
-              maxX = x;
-
-            double y = row.arg2().evalDouble();
-            if (y < minY)
-              minY = y;
-            if (y > maxY)
-              maxY = y;
-
-            double z = row.arg3().evalDouble();
-            if (z < minZ)
-              minZ = z;
-            if (z > maxZ)
-              maxZ = z;
-          } catch (ArgumentTypeException ate) {
-            // ignore this row
-          }
-        }
+  private IExpr heightLinePlot(IAST heights, IAST plotStyle, EvalEngine engine) {
+    final int valuesSize = heights.size();
+    IASTAppendable resultList = F.ListAlloc(valuesSize);
+
+    IExpr flattenHeights = engine.evaluate(F.Flatten((IExpr) heights));
+    double deltaHeight =
+        engine.evaluate(F.Max(flattenHeights).subtract(F.Min(flattenHeights))).evalDouble();
+
+    int lineColorNumber = 1;
+
+    for (int i = 1; i < valuesSize; i++) {
+      IAST rowList = (IAST) heights.get(i);
+      final int rowListSize = rowList.size();
 
+      IASTAppendable lineList = F.ListAlloc(rowListSize);
+
+      for (int j = 1; j < rowListSize; j++) {
         // ListLinePlot3D size is 2.5 × 2.5 × 1 independently from its coordinates
-        IAST deltaXYZ =
-            F.List(F.num((maxX - minX) / 2.5), F.num((maxY - minY) / 2.5), F.num(maxZ - minZ));
+        lineList.append(F.List(F.num(i * 2.5 / valuesSize), F.num(j * 2.5 / rowListSize),
```

Comment:
If you new you have only `double` elements in your list, you can write it directly into a `F.List(...)`.

Reply:
This one isn't.